### ISSUE TITLE
Object mask difficulty

### ISSUE THREAD

**Issue by wrpooley** [blakeblackshear/frigate#1583]
I'm having a tough time implementing an object mask.  After getting the coordinates from the creator and editing the config file (see below) and after restarting Frigate there is no object mask in place.  It's probably a yam thing but I've tried many times with no luck.  Anyone see the issue here?  I have no trouble adding a motion mask.

```yaml
  Driveway:
    ffmpeg:
      hwaccel_args:
        - -hwaccel_output_format
        - yuv420p
      input_args:
        - -avoid_negative_ts
        - make_zero

      inputs:
        - path: rtmp://192.168.111.50/bcs/channel0_sub.bcs?channel=0&stream=0&user=&password
          roles:
            - detect
            - clips

            
    width: 640
    height: 480
    fps: 5
    objects:
      track:
        - person
      filters:
        person:
          mask: 
            - 214,300,277,311,343,281,242,236
    snapshots:
      enabled: true
      retain:
        default: 2
    clips:
      enabled: true
      retain:
        default: 2
    motion:
      mask:
        - 229,465,19,466,21,448,223,449
        - 474,167,226,67,0,180,0,128,0,108,0,69,0,0,257,0,455,0,640,0,640,134,640,249
```

**Comment by blakeblackshear** [comment]
How do you know the object mask is not in place?

**Comment by wrpooley** [comment]
I’m not in front of the camera at the moment to test it so I suppose I can’t confirm it that way. Presumably object mask coordinates would persist in the Mask generator UI after restarting the same way the motion masks do. However they do not.

**Comment by blakeblackshear** [comment]
Check the compiled JSON config under the debug tab. Frigate won't even load if the config doesn't parse correctly. It doesn't discard invalid parts of the config. Are you sure you are updating the right file?

Also, keep in mind that object masks are based on the bottom center of the bounding box. It doesn't matter if other parts of the bounding box overlap with the object mask.

**Comment by wrpooley** [comment]
Thanks for the response! I’m currently using frigate as an add on to HomeAssistant and I’m using the native mobile app. If I understand your question correctly I can see the JSON file under the Debug window which I’ve attached to this message. The camera in question is the driveway camera. Also, the file I’m  updating with the coordinates  is the frigate.yaml 

PS. The success of this add on has single-handedly walked my wife back from the ledge of throwing our 4 cam reolink NVR platform in a trashcan and buying a ring system! Thank you for that. There’s still some things to work out but all in all a great experience. 

`{
  "cameras": {
    "Back_Porch": {
      "best_image_timeout": 60,
      "clips": {
        "enabled": true,
        "objects": null,
        "post_capture": 5,
        "pre_capture": 5,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        }
      },
      "detect": {
        "enabled": true,
        "max_disappeared": 25
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -c:v h264_v4l2m2m -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -i rtmp://192.168.111.50/bcs/channel2_sub.bcs?channel=0&stream=0&user=&password=-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Back_Porch-%Y%m%d%H%M%S.mp4 -r 5 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect",
            "clips"
          ]
        }
      ],
      "fps": 5,
      "frame_shape": [
        480,
        640
      ],
      "height": 480,
      "motion": {
        "contour_area": 100,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 80,
        "mask": [
          "640,480,475,480,502,428,438,400,462,310,536,315,561,204,563,97,640,96,640,315",
          "225,480,0,480,0,447,225,448",
          "0,0,0,73,137,0"
        ],
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "required_zones": [],
        "timestamp": true
      },
      "name": "Back_Porch",
      "objects": {
        "filters": {
          "person": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": null,
        "track": [
          "person"
        ]
      },
      "record": {
        "enabled": false,
        "retain_days": 30
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": false,
        "crop": false,
        "enabled": true,
        "height": null,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        },
        "timestamp": false
      },
      "width": 640,
      "zones": {}
    },
    "Driveway": {
      "best_image_timeout": 60,
      "clips": {
        "enabled": true,
        "objects": null,
        "post_capture": 5,
        "pre_capture": 5,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        }
      },
      "detect": {
        "enabled": true,
        "max_disappeared": 25
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -i rtmp://192.168.111.50/bcs/channel0_sub.bcs?channel=0&stream=0&user=&password=-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Driveway-%Y%m%d%H%M%S.mp4 -r 5 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect",
            "clips"
          ]
        }
      ],
      "fps": 5,
      "frame_shape": [
        480,
        640
      ],
      "height": 480,
      "motion": {
        "contour_area": 100,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 80,
        "mask": [
          "229,465,19,466,21,448,223,449",
          "474,167,226,67,0,180,0,128,0,108,0,69,0,0,257,0,455,0,640,0,640,134,640,249"
        ],
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "required_zones": [],
        "timestamp": true
      },
      "name": "Driveway",
      "objects": {
        "filters": {
          "person": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": null,
        "track": [
          "person"
        ]
      },
      "record": {
        "enabled": false,
        "retain_days": 30
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": false,
        "crop": false,
        "enabled": true,
        "height": null,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        },
        "timestamp": false
      },
      "width": 640,
      "zones": {}
    },
    "Lumus": {
      "best_image_timeout": 60,
      "clips": {
        "enabled": true,
        "objects": null,
        "post_capture": 5,
        "pre_capture": 5,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        }
      },
      "detect": {
        "enabled": true,
        "max_disappeared": 25
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -c:v h264_v4l2m2m -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -i rtmp://192.168.111.50/bcs/channel1_sub.bcs?channel=0&stream=0&user=&password=-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Lumus-%Y%m%d%H%M%S.mp4 -r 5 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect",
            "clips"
          ]
        }
      ],
      "fps": 5,
      "frame_shape": [
        480,
        704
      ],
      "height": 480,
      "motion": {
        "contour_area": 100,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 80,
        "mask": [
          "201,457,365,459,368,425,202,425"
        ],
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "required_zones": [],
        "timestamp": true
      },
      "name": "Lumus",
      "objects": {
        "filters": {
          "person": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": null,
        "track": [
          "person"
        ]
      },
      "record": {
        "enabled": false,
        "retain_days": 30
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": false,
        "crop": false,
        "enabled": true,
        "height": null,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        },
        "timestamp": false
      },
      "width": 704,
      "zones": {}
    },
    "Side_Porch": {
      "best_image_timeout": 60,
      "clips": {
        "enabled": true,
        "objects": null,
        "post_capture": 5,
        "pre_capture": 5,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        }
      },
      "detect": {
        "enabled": true,
        "max_disappeared": 25
      },
      "ffmpeg_cmds": [
        {
          "cmd": "ffmpeg -hide_banner -loglevel warning -c:v h264_v4l2m2m -hwaccel_output_format yuv420p -avoid_negative_ts make_zero -i rtmp://192.168.111.50/bcs/channel3_sub.bcs?channel=0&stream=0&user=&password=-f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/Side_Porch-%Y%m%d%H%M%S.mp4 -r 5 -f rawvideo -pix_fmt yuv420p pipe:",
          "roles": [
            "detect",
            "clips"
          ]
        }
      ],
      "fps": 5,
      "frame_shape": [
        480,
        640
      ],
      "height": 480,
      "motion": {
        "contour_area": 100,
        "delta_alpha": 0.2,
        "frame_alpha": 0.2,
        "frame_height": 80,
        "mask": [
          "190,450,181,473,106,473,108,452"
        ],
        "threshold": 25
      },
      "mqtt": {
        "bounding_box": true,
        "crop": true,
        "enabled": true,
        "height": 270,
        "required_zones": [],
        "timestamp": true
      },
      "name": "Side_Porch",
      "objects": {
        "filters": {
          "person": {
            "mask": [],
            "max_area": 24000000,
            "min_area": 0,
            "min_score": 0.5,
            "threshold": 0.7
          }
        },
        "mask": null,
        "track": [
          "person"
        ]
      },
      "record": {
        "enabled": false,
        "retain_days": 30
      },
      "rtmp": {
        "enabled": true
      },
      "snapshots": {
        "bounding_box": false,
        "crop": false,
        "enabled": true,
        "height": null,
        "required_zones": [],
        "retain": {
          "default": 2,
          "objects": {}
        },
        "timestamp": false
      },
      "width": 640,
      "zones": {}
    }
  },
  "clips": {
    "max_seconds": 300,
    "retain": {
      "default": 10,
      "objects": {}
    },
    "tmpfs_cache_size": ""
  },
  "database": {
    "path": "/media/frigate/clips/frigate.db"
  },
  "detectors": {
    "coral": {
      "device": "usb",
      "num_threads": 3,
      "type": "edgetpu"
    }
  },
  "environment_vars": {},
  "logger": {
    "default": "INFO",
    "logs": {}
  },
  "model": {
    "height": 320,
    "width": 320
  },
  "mqtt": {
    "client_id": "frigate",
    "host": "192.168.111.200",
    "port": 1883,
    "stats_interval": 60,
    "topic_prefix": "frigate",
    "user": "xxx"
  },
  "snapshots": {
    "retain": {
      "default": 10,
      "objects": {}
    }
  }
} `

**Comment by blakeblackshear** [comment]
The object mask for Driveway is clearly empty in that config. Are you certain you are modifying the right file? You did restart the addon correct?

**Comment by wrpooley** [comment]
It seems to be something unusual about this particular camera.  I tried creating an object mask on any/all of the other 3 cameras and it works as expected.   In my house I have 2 Reolink RLC 510A (wired), 1 Reolink Lumus (wireless) and 1 RLC 511W (wireless).  This last cam is the one that won't accept the object mask (or at least doesn't show that it has).  All the cameras are configured the same except the RLC 511W does not have hardware acceleration turned on.  I found through trial and error that was the only way to make it work.  I wonder if that has anything to do with it?
